demos/ping_pong.py

```diff
@@ -0,0 +1,233 @@
+# unlike the `interpolate` approach. However, contrary to the `interpolate` approach
+# not introducing new extrema, the `project` approach can be seen to introduce new
+# minimum values. No new maximum values are introduced, although the maximum values do
+# decrease slightly.
+#
+# To summarise, the `project` approach:
+#
+# 1. It is conservative.
+# 2. It may introduce new extrema.
+#
+# Note that neither of the approaches considered thus far provide an interpolation
+# method that is both conservative and does not introduce new extrema. In the case of
+# :math:`\mathbb P1` fields specifically, it is possible to achieve this using 'mass
+# lumping'. Recall the linear system above. Lumping amounts to replacing the mass matrix
+# :math:`\underline{\mathbf{M}_B}` with a diagonal matrix, whose entries correspond to
+# the sums over the corresponding mass matrix rows. (See :cite:`FPP+:2009` for details.)
+#
+# The resulting bounded projection operator can be used in Animate by passing
+# `bounded=True` to the `project` function. ::
+
+quantities["integral"]["bounded"] = [initial_integral]
+quantities["minimum"]["bounded"] = [initial_min]
+quantities["maximum"]["bounded"] = [initial_max]
+f_bounded = Function(V_A).assign(sensor)
+for _ in range(niter):
+    f_bounded = project(project(f_bounded, V_B, bounded=True), V_A, bounded=True)
+    quantities["integral"]["bounded"].append(assemble(f_bounded * dx))
+    quantities["minimum"]["bounded"].append(f_bounded.vector().gather().min())
+    quantities["maximum"]["bounded"].append(f_bounded.vector().gather().max())
+
+fig, axes = plt.subplots(ncols=3, figsize=(15, 5))
+for i, (key, subdict) in enumerate(quantities.items()):
+    axes[i].plot(subdict["interpolate"], label="Interpolate")
+    axes[i].plot(subdict["project"], label="Project")
+    axes[i].plot(subdict["bounded"], label="Bounded project")
+    axes[i].set_xlabel("Iteration")
+    axes[i].set_ylabel(key.capitalize())
+    axes[i].grid(True)
+axes[1].legend()
+plt.savefig("ping_pong-quantities_bounded.jpg")
+
+
+# To check that the interpolants still resemble the sensor function after 50 iterations,
+# we plot the final fields alongside it. ::
+
+fig, axes = plt.subplots(ncols=2, nrows=2, figsize=(10, 10))
+colourbars = []
+ax = axes[0][0]
+colourbars.append(fig.colorbar(tricontourf(sensor, axes=ax), ax=ax))
+ax.set_title("Source function")
+ax = axes[0][1]
+colourbars.append(fig.colorbar(tricontourf(f_interp, axes=ax), ax=ax))
+ax.set_title("Interpolate")
+ax = axes[1][0]
+colourbars.append(fig.colorbar(tricontourf(f_proj, axes=ax), ax=ax))
+ax.set_title("Project")
+ax = axes[1][1]
+colourbars.append(fig.colorbar(tricontourf(f_bounded, axes=ax), ax=ax))
+ax.set_title("Bounded")
+for i in range(2):
+    for j in range(2):
+        axes[i][j].axis(False)
+        axes[i][j].set_aspect(1)
+for cbar in colourbars:
+    cbar.set_ticks([-0.05, 0.15, 0.3, 0.45, 0.6, 0.75, 0.9, 1.05])
```

Comment:
I left a separate comment on how to deal with colourbars.

I also think it would be nice to add a contour at 0., so that we can actually see in the `project` figure that it introduces a new minimum.

Reply:
And I'd like to remove the 1.05 tick to remove any suggestion that there might be any values >1.